src/extras-vectors.js

```
@@ -0,0 +1,66 @@
+const fetch = require('node-fetch').default;
+
+/**
+ * Gets the vector for the given text from SillyTavern-extras
+ * @param {string[]} texts - The array of texts to get the vector for
+ * @param {string} apiUrl - The Extras API URL
+ * @param {string} - The Extras API key, or empty string if API key not enabled
+ * @returns {Promise<number[][]>} - The array of vectors for the texts
+ */
+async function getExtrasBatchVector(texts, apiUrl, apiKey) {
+    return getExtrasVector(texts, apiUrl, apiKey);  // The implementation supports batches transparently.
+}
+
+module.exports = {
+    getExtrasVector,
+    getExtrasBatchVector,
+};
+
+/**
+ * Gets the vector for the given text from SillyTavern-extras
+ * @param {string|string[]} text - The text or texts to get the vector for
+ * @param {string} apiUrl - The Extras API URL
+ * @param {string} - The Extras API key, or empty string if API key not enabled
+ * @returns {Promise<number[]>|Promise<number[][]>} - The vector for a single text, or the array of vectors for multiple texts
+ */
+async function getExtrasVector(text, apiUrl, apiKey) {
+    let url;
+    try {
+        url = new URL(apiUrl);
+        url.pathname = '/api/embeddings/compute';
+    }
+    catch (error) {
+        console.log('Failed to set up Extras API call:', error);
+        console.log('Extras API URL given was:', apiUrl);
```

You can rethrow an error here not to attempt a fetch if the URL is invalid.

---

Thanks for the catch. Meant to do that, but forgot to throw.